snuba/datasets/discover.py

```diff
@@ -264,7 +282,24 @@ def get_query_processors(self) -> Sequence[QueryProcessor]:
         discover_source = self.get_dataset_schemas().get_read_schema().get_data_source()
 
         return [
-            DatasetSelector(discover_source, self.__events_columns, self.__transactions_columns),
+            DatasetSelector(
+                discover_source=discover_source,
+                events_columns=self.__events_columns,
+                transactions_columns=self.__transactions_columns,
+                followup_processors={
```

Comment:
There are fundamentally two ways to do this:
1) The DatasetSelector processor just picks the dataset and the NestedFieldConditionOptimizer try to figure out (from the tablesource picked by the DatasetSelector) whether the source supports the flattened_tag column. I don't like this approach because it hides a dependency between the two processor that actually exists.
2) The dataset selector picks the table source and then schedules the followup processing that is specifically a consequence of the dataset selection decision. Which is what I am doing here. This way the path between processor is evident and visible here.

I think this mechanism should become generic and supported by the framework. That's a followup work to do.

Reply:
I think that's fine as long as a processor will only need to be run directly after the one it is registered as a follow up of, and not scheduled for some later stage

Reply:
"not scheduled for some later stage" we should try to stay a way from this. It would massively increase the complexity of the processor system and make it impossible to have a big picture of what is operating on the query.